docs/billing-integration.md

```diff
@@ -0,0 +1,273 @@
+
+The reporting secret contains the following three fields (see an [example](https://storage.cloud.google.com/cloud-marketplace-tools/reporting_secrets/fake_reporting_secret.yaml)):
+* **consumer-id** - an identifier that represents the customer being billed for usage. To be passed verbatim to Google Service Control.
+* **entitlement-id** - another identifier representing the customer, but generally unused.
+* **reporting-key** - a base64-encoded GCP JSON service account key to use when reporting usage.
+
+**Note:** when viewing the YAML representation of this secret, every field is base64-encoded and will be automatically decoded by Kubernetes when reading values. The **reporting-key** field is encoded an additional time, so your application will need to decode it (after it is automatically decoded by Kubernetes once) if it needs to read the actual JSON key data.
+
+## Using the Metering Agent
+
+You can choose to use the Marketplace-provided [metering agent](https://github.com/GoogleCloudPlatform/ubbagent) to simplify usage reporting. The metering agent can be included into your Kubernetes deployment as a sidecar container, included in your own container, or built directly into your application if it's written in Go. Including the agent as a sidecar is the most straightforward.
+
+The metering agent is configured with usage reporting credentials, the metrics that will be reported, and the report destination. Much of the configuration is boilerplate, and you'll need to only update some identifiers related to your application.
+
+If your application is priced by usage time (how long a deployment runs), you can configure the agent to report automatically. For anything else your application will need to report usage to the agent either periodically or as it occurs. When deployed as a sidecar, the metering agent listens on the loopback interface (127.0.0.1), and is only reachable by other containers in the same pod.
+
+**Note:** the metering agent's HTTP interface is unencrypted and unauthenticated, and works under the assumption that access is limited to containers within the same pod.
+
+### Agent Deployment and Configuration
+
+The following excerpts include the metering agent as a sidecar along with some application container. The agent's configuration is included as a ConfigMap value. When deployed as a sidecar, the agent's configuration can be parameterized through the use of environment variables, which is the mechanism used for passing values from the reporting secret.
+
+Kubernetes manifest excerpt:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: $name-ubbagent-config
+data:
+  config.yaml: |
+    # The identity section contains authentication information used
+    # by the agent.
+    identities:
+    - name: gcp
+      gcp:
+        # This parameter accepts a base64-encoded JSON service
+        # account key. The value comes from the reporting secret.
+        encodedServiceAccountKey: $AGENT_ENCODED_KEY
```

Comment:
I don't remember if this will get substituted with empty string by `envsubst`. I _believe_ we fixed it so that unrecognized variables don't get substituted.

Regardless, I think it's a good idea to highlight that this isn't intended to be substituted. Maybe changing `$name` to something helm-like `{{ .name }}`? We should change all env variables in this snippet as well.

Reply:
Added an explanation.